spec/validation/configure_spec.rb

```diff
@@ -34,244 +41,264 @@
+      h = correct_hash.deep_merge(self: [{
+                                    question: 'I have no identifier',
+                                    identifier: '',
+                                  }])
+      expect_configure_error(h, /must be filled/)
     end
 
-    it 'fails if question if the title is empty' do
-      h = correct_hash.deep_merge(node: {
-                                    missing_title_question: {
-                                      question: '',
-                                      default: 'I am missing my question!',
-                                    },
-                                  })
-      results = run_configure_validation(h)
-      expect(results[:parameters][:question][0]).to eq('must be filled')
+    it 'fails if question is missing or empty' do
+      h = correct_hash.deep_merge(self: [{ identifier: 'missing_question' }])
+      expect_configure_error(h, /is missing/)
+      h = correct_hash.deep_merge(self: [{
+                                    question: '',
+                                    identifier: 'no_question',
+                                  }])
+      expect_configure_error(h, /must be filled/)
     end
 
     it "fails if type isn't supported" do
-      h = correct_hash.deep_merge(group: {
-                                    unsupported_type: {
-                                      question: 'Do I have an unsupported type?',
-                                      type: 'Unsupported',
-                                    },
-                                  })
-      results = run_configure_validation(h)
-      expect(results[:parameters].keys).to eq([:type])
+      h = correct_hash.deep_merge(group: [{
+                                    identifier: 'unsupported_type',
+                                    question: 'Do I have an unsupported type?',
+                                    type: 'Unsupported',
+                                  }])
+      expect_configure_error(h, /Is an unsupported question type/)
     end
 
     it 'fails if the optional input is not true or false' do
-      h = correct_hash.deep_merge(group: {
-                                    invalid_optional_flag: {
-                                      question: 'Do I have a boolean optional input?',
-                                      optional: 'I should be true or false',
-                                    },
-                                  })
-      results = run_configure_validation(h)
-      expect(results[:parameters].keys).to eq([:optional])
+      h = correct_hash.deep_merge(group: [{
+                                    identifier: 'invalid_optional_flag',
+                                    question: 'Do I have a boolean optional input?',
+                                    optional: 'I should be true or false',
+                                  }])
+      expect_configure_error(h, /must be boolean/)
     end
   end
 
   context 'with missing question blocks' do
```

Comment:
You might be able to de-duplicate these with something like (untested):

```ruby
[:domain, :group, :node, :self].each do |question_type|
  it "fails when #{question_type} is missing" do
    correct_hash.delete(question_type)
    expect_configure_error(correct_hash, /is missing/)
  end
end
```

You may or may not think this is worth doing.

Reply:
Fixed: https://github.com/alces-software/metalware/pull/185/commits/f29c53c975f72f50e204930a79cad7d34ce093db